crates/starknet-devnet-core/src/starknet/get_class_impls.rs

```diff
@@ -29,8 +29,21 @@ pub fn get_class_impl(
     block_id: &BlockId,
     class_hash: ClassHash,
 ) -> DevnetResult<ContractClass> {
-    let state = starknet.get_mut_state_at(block_id)?;
-    match state.get_rpc_contract_class(&class_hash) {
+    // if provided with block hash, convert to number - the underlying logic only works with that or
+    // block tag
+    let block_id = if let BlockId::Hash(block_hash) = block_id {
```

Comment:
So that we can query a class storage with `N` classes in `O(1)` by relying on the `if storage_block_number <= query_block_number` trick. Otherwise, for each query we would need to traverse all class commitments, leading to `O(N)`.

Reply:
ok got it but still not sure how it will impact block abortion feature